onadata/libs/filters.py

```diff
@@ -361,6 +361,9 @@ def _xform_filter(self, request, view, keyword):
 
         if request.user.is_anonymous:
             xforms = xform_qs.filter(shared_data=True)
+        elif public_export:
+            export = get_object_or_404(Export, pk=view.kwargs.get("pk"))
+            xforms = XForm.objects.filter(pk=export.xform.pk)
```

Comment:
Can we go with `export.xform_id` since `export.xform.pk` has a tendency to result in another hit to the DB that can be avoided here?

```
 xforms = XForm.objects.filter(pk=export.xform_id)
```